luaclang.c

```diff
@@ -104,6 +106,111 @@ static int cursor_getspelling(lua_State *L)
+    switch (kind) {
+        case CXCursor_StructDecl: 
+                return "StructDecl";
+        case CXCursor_UnionDecl: 
+                return "UnionDecl";
+        case CXCursor_EnumDecl: 
+                return "EnumDecl";
+        case CXCursor_FieldDecl: 
+                return "FieldDecl";
+        case CXCursor_EnumConstantDecl: 
+                return "EnumConstantDecl";
+        case CXCursor_FunctionDecl: 
+                return "FunctionDecl";
+        case CXCursor_VarDecl: 
+                return "VarDecl";
+        case CXCursor_ParmDecl: 
+                return "ParmDecl";
+        case CXCursor_TypedefDecl: 
+                return "TypedefDecl";
+        case CXCursor_IntegerLiteral: 
+                return "IntegerLiteral";
+        default: 
+                return "Unaddressed";
+    }
+}
+
+/*      
+        Format - cur:getKind()
+        Parameter - cur - Cursor whose kind is to be obtained    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__MANIP.html#ga018aaf60362cb751e517d9f8620d490c
+*/
+static int cursor_getkind(lua_State *L)
+{
+        CXCursor *cur;
+        to_object(L, cur, CURSOR_METATABLE, 1);
+        lua_pushstring(L, cursor_kind_str(clang_getCursorKind(*cur)));
+        return 1;
+}
+
+/* 
+        Visitor function for visit_children
+        A value of this enumeration type should be returned by this functionr to indicate how visit_children proceed : 
+                1. CXChildVisit_Break - Terminates the cursor traversal 
+                2. CXChildVisit_Continue - Continues the cursor traversal with the next sibling of the cursor just visited, without visiting its children.
+                3. CXChildVisit_Recurse - Recursively traverse the children of this cursor, using the same visitor and client data.
+ */
+enum CXChildVisitResult visitor_function(CXCursor cursor, CXCursor parent, CXClientData client_data)
+{
+        lua_State *L = (lua_State*) client_data;
+        lua_pushvalue(L, 1);    
+        CXCursor *cur;
+        new_object(L, cur, CURSOR_METATABLE);           
+        *cur = cursor;
+        CXCursor *par;
+        new_object(L, par, CURSOR_METATABLE);
+        *par = parent;
+        if (lua_pcall(L, 2, 1, 0) != 0) {
+                lua_settop(L, 0);
+                lua_pushstring(L, "Function call unsuccessfull");
+                return CXChildVisit_Break;
+        }
+        const char *result = lua_tostring(L, -1);
+        if (strcmp(result, "continue") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Continue;
+        }
+        else if (strcmp(result, "recurse") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Recurse;
+        }
+        else if (strcmp(result, "break") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Break;
+        }
+        else {
+                lua_settop(L, 0);
+                lua_pushstring(L, "Undefined return to visitor");
```

Comment:
Avoid capitalising error messages.
```suggestion
                lua_pushstring(L, "undefined return to visitor");
```